src/common/wui_api.c

```diff
@@ -7,5 +7,104 @@
  */
 #include "wui_api.h"
 
-void get_eth_params(wui_eth_params_s eth_params) {
+#include "ini_handler.h"
+#include "eeprom.h"
+#include "string.h"
+
+#define LAN_MSK_TYPE      2 //EEPROM mask for user-defined settings (Switch between dhcp and static)
+#define IP4_TYPE_FLAG_POS 1
+
+#define IS_LAN_STATIC(flg) (flg & LAN_MSK_TYPE)
+#define IS_LAN_DHCP(flg)   (!IS_LAN_STATIC(flg))
+
+static int ini_handler_func(void *user, const char *section, const char *name, const char *value) {
+
+    ETH_Config_t *tmp_config = (ETH_Config_t *)user;
+
+#define MATCH(s, n) strcmp(section, s) == 0 && strcmp(name, n) == 0
+
+    if (MATCH("lan_ip4", "type")) {
+        if (strncmp(value, "DHCP", 4) == 0 || strncmp(value, "dhcp", 4) == 0) {
+            tmp_config->lan_flag &= ~LAN_EEFLG_TYPE;
+            tmp_config->set_flag |= NETVAR_MSK(NETVAR_LAN_FLAGS);
+        } else if (strncmp(value, "STATIC", 6) == 0 || strncmp(value, "static", 6) == 0) {
+            tmp_config->lan_flag |= LAN_EEFLG_TYPE;
+            tmp_config->set_flag |= NETVAR_MSK(NETVAR_LAN_FLAGS);
+        }
+    } else if (MATCH("lan_ip4", "hostname")) {
+        strlcpy(tmp_config->hostname, value, LAN_HOSTNAME_MAX_LEN + 1);
+        tmp_config->hostname[LAN_HOSTNAME_MAX_LEN] = '\0';
+        tmp_config->set_flag |= NETVAR_MSK(NETVAR_HOSTNAME);
+    } else if (MATCH("lan_ip4", "address")) {
+        if (ip4addr_aton(value, &tmp_config->lan_ip4_addr)) {
+            tmp_config->set_flag |= NETVAR_MSK(NETVAR_LAN_IP4_ADDR);
+        }
+    } else if (MATCH("lan_ip4", "mask")) {
+        if (ip4addr_aton(value, &tmp_config->lan_ip4_msk)) {
+            tmp_config->set_flag |= NETVAR_MSK(NETVAR_LAN_IP4_MSK);
+        }
+    } else if (MATCH("lan_ip4", "gateway")) {
+        if (ip4addr_aton(value, &tmp_config->lan_ip4_gw)) {
+            tmp_config->set_flag |= NETVAR_MSK(NETVAR_LAN_IP4_GW);
+        }
+    } else if (MATCH("connect", "address")) {
+        if (ip4addr_aton(value, &tmp_config->connect_ip4)) {
+            tmp_config->set_flag |= NETVAR_MSK(NETVAR_CONNECT_IP4);
+        }
+    } else if (MATCH("connect", "token")) {
+        strlcpy(tmp_config->connect_token, value, CONNECT_TOKEN_SIZE + 1);
+        tmp_config->connect_token[CONNECT_TOKEN_SIZE] = '\0';
+        tmp_config->set_flag |= NETVAR_MSK(NETVAR_CONNECT_TOKEN);
+    } else {
+        return 0; /* unknown section/name, error */
+    }
+    return 1;
+}
+
+static ini_handler wui_init_handler = ini_handler_func;
+
+void load_ini_params(ETH_Config_t *config_t) {
+    ini_load_file(wui_init_handler, (void *)config_t);
+}
+
+uint32_t eeprom_set_eth_params(ETH_Config_t *config_t) {
```

Comment:
This function is probably not enough. There are cases that wants to set only LAN flag and that is not possible here.